test/unit/scval_test.js

```diff
@@ -206,4 +210,31 @@ describe('parsing and building ScVals', function () {
   it('throws on arrays with mixed types', function () {
     expect(() => nativeToScVal([1, 'a', false])).to.throw(/same type/i);
   });
+
+  it('lets strings be large integer ScVals', function () {
+    ['i64', 'i128', 'i256', 'u64', 'u128', 'u256'].forEach((type) => {
+      const scv = nativeToScVal('12345', { type });
+      expect(XdrLargeInt.getType(scv.switch().name)).to.equal(type);
```

Comment:
is it worth adding a line or separate test case, to demonstrate expectation when invalid
```
XdrLargeInt.getType('ab');
```

Reply:
good call! but this is implicitly already handled by an existing test case, here:

https://github.com/stellar/js-stellar-base/blob/25a2b68a7dd99c182edf74f46bd75a998d59e7bd/test/unit/scint_test.js#L247-L251